### ISSUE TITLE
cublas runtime error : resource allocation failed

### ISSUE THREAD

**Issue by Rainelz** [allenai/allennlp#2636]
EDIT/SOLVED: dumb problem. the tokenizer at some point was returning single token Instances due to weird data preprocessing. Feel free to delete

**Describe the bug**
Hi!
I am trying to create a very basic dialogue generator and trying some configs on a known dataset, I've been playing with the seq2seq model by modifying the languagemodeling reader to return source-target keys as expected by the model and everything was running. 
Yesterday, I tried to specify a "tokens_per_instance" value to do some tests with a lot of different parameters but when I went back to the old configuration this started happening:

 

> 13:07:58,306 - INFO - allennlp.training.trainer - Training
>   0%|          | 0/36 [00:00<?, ?it/s]/opt/conda/conda-bld/pytorch_1549628766161/work/aten/src/THC/THCTensorScatterGather.cu:97: void THCudaTensor_gatherKernel(TensorInfo<Real, IndexType>, TensorInfo<Real, IndexType>, TensorInfo<long, IndexType>, int, IndexType) [with IndexType = unsigned int, Real = float, Dims = 3]: block: [12,0,0], thread: [96,0,0] Assertion `indexValue >= 0 && indexValue < src.sizes[dim]` failed.
> /opt/conda/conda-bld/pytorch_1549628766161/work/aten/src/THC/THCTensorScatterGather.cu:97: void THCudaTensor_gatherKernel(TensorInfo<Real, IndexType>, TensorInfo<Real, IndexType>, TensorInfo<long, IndexType>, int, IndexType) [with IndexType = unsigned int, Real = float, Dims = 3]: block: [12,0,0], thread: [97,0,0] Assertion `indexValue >= 0 && indexValue < src.sizes[dim]` failed.
> ....
> /opt/conda/conda-bld/pytorch_1549628766161/work/aten/src/THC/THCTensorScatterGather.cu:97: void THCudaTensor_gatherKernel(TensorInfo<Real, IndexType>, TensorInfo<Real, IndexType>, TensorInfo<long, IndexType>, int, IndexType) [with IndexType = unsigned int, Real = float, Dims = 3]: block: [12,0,0], thread: [511,0,0] Assertion `indexValue >= 0 && indexValue < src.sizes[dim]` failed.
> Traceback (most recent call last):
>   File "/home/rainelz/miniconda3/envs/NLPTEST/bin/allennlp", line 10, in <module>
>     sys.exit(run())
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/run.py", line 18, in run
>     main(prog="allennlp")
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/commands/__init__.py", line 101, in main
>     args.func(args)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/commands/train.py", line 103, in train_model_from_args
>     args.force)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/commands/train.py", line 136, in train_model_from_file
>     return train_model(params, serialization_dir, file_friendly_logging, recover, force)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/commands/train.py", line 204, in train_model
>     metrics = trainer.train()
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/training/trainer.py", line 465, in train
>     train_metrics = self._train_epoch(epoch)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/training/trainer.py", line 306, in _train_epoch
>     loss = self.batch_loss(batch_group, for_training=True)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/training/trainer.py", line 247, in batch_loss
>     output_dict = self.model(**batch)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/torch/nn/modules/module.py", line 489, in __call__
>     result = self.forward(*input, **kwargs)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/models/encoder_decoders/simple_seq2seq.py", line 218, in forward
>     output_dict = self._forward_loop(state, target_tokens)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/models/encoder_decoders/simple_seq2seq.py", line 340, in _forward_loop
>     output_projections, state = self._prepare_output_projections(input_choices, state)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/models/encoder_decoders/simple_seq2seq.py", line 427, in _prepare_output_projections
>     (decoder_hidden, decoder_context))
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/torch/nn/modules/module.py", line 489, in __call__
>     result = self.forward(*input, **kwargs)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/torch/nn/modules/rnn.py", line 732, in forward
>     self.bias_ih, self.bias_hh,
> RuntimeError: cublas runtime error : resource allocation failed at /opt/conda/conda-bld/pytorch_1549628766161/work/aten/src/THC/THCGeneral.cpp:250

This is the config I'm using:

> `{
>     "dataset_reader": {
>         "type": "language_modeling",
>         "tokenizer": {
>             "type": "word",
>             "word_splitter": {
>                 "type": "spacy"
>             }
>         }
>     },
>     "iterator": {
>         "type": "basic",
>         "batch_size": 128
>     },
>     "model": {
>         "type": "simple_seq2seq",
>         "encoder": {
>             "type": "lstm",
>             "bidirectional": true,
>             "dropout": 0.2,
>             "hidden_size": 256,
>             "input_size": 200,
>             "num_layers": 2
>         },
>         "max_decoding_steps": 20,
>         "source_embedder": {
>             "type": "basic",
>             "tokens": {
>                 "embedding_dim": 200
>             }
>         }
>     },
>     "train_data_path": "/home/rainelz/Data/moes_tavern_processed.txt",
>     "trainer": {
>         "cuda_device": 0,
>         "grad_clipping": 5,
>         "num_epochs": 100,
>         "optimizer": "adam",
>         "patience": 10
>     }
> }`

this is the currently working config:

> `{
>     "dataset_reader": {
>         "type": "language_modeling",
>         "tokenizer": {
>             "type": "word",
>             "word_splitter": {
>                 "type": "spacy"
>             }
>         },
>         "tokens_per_instance": 50
>     },
>     "iterator": {
>         "type": "basic",
>         "batch_size": 128
>     },
>     "model": {
>         "type": "simple_seq2seq",
>         "encoder": {
>             "type": "lstm",
>             "bidirectional": true,
>             "dropout": 0.2,
>             "hidden_size": 256,
>             "input_size": 1000,
>             "num_layers": 2
>         },
>         "max_decoding_steps": 20,
>         "source_embedder": {
>             "type": "basic",
>             "tokens": {
>                 "embedding_dim": 1000
>             }
>         }
>     },
>     "train_data_path": "/home/rainelz/Data/moes_tavern_processed.txt",
>     "trainer": {
>         "cuda_device": 0,
>         "grad_clipping": 5,
>         "num_epochs": 100,
>         "optimizer": "adam",
>         "patience": 10
>     }
> }`

if I run with CUDA_LAUNCH_BLOCKING=1 I get:

> Traceback (most recent call last):
>   File "/home/rainelz/miniconda3/envs/NLPTEST/bin/allennlp", line 10, in <module>
>     sys.exit(run())
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/run.py", line 18, in run
>     main(prog="allennlp")
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/commands/__init__.py", line 101, in main
>     args.func(args)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/commands/train.py", line 103, in train_model_from_args
>     args.force)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/commands/train.py", line 136, in train_model_from_file
>     return train_model(params, serialization_dir, file_friendly_logging, recover, force)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/commands/train.py", line 204, in train_model
>     metrics = trainer.train()
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/training/trainer.py", line 465, in train
>     train_metrics = self._train_epoch(epoch)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/training/trainer.py", line 306, in _train_epoch
>     loss = self.batch_loss(batch_group, for_training=True)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/training/trainer.py", line 247, in batch_loss
>     output_dict = self.model(**batch)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/torch/nn/modules/module.py", line 489, in __call__
>     result = self.forward(*input, **kwargs)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/models/encoder_decoders/simple_seq2seq.py", line 215, in forward
>     state = self._init_decoder_state(state)
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/models/encoder_decoders/simple_seq2seq.py", line 284, in _init_decoder_state
>     self._encoder.is_bidirectional())
>   File "/home/rainelz/miniconda3/envs/NLPTEST/lib/python3.6/site-packages/allennlp/nn/util.py", line 196, in get_final_encoder_states
>     final_encoder_output = encoder_outputs.gather(1, expanded_indices)
> RuntimeError: cuda runtime error (59) : device-side assert triggered at /opt/conda/conda-bld/pytorch_1549628766161/work/aten/src/THC/generic/THCTensorScatterGather.cu:71
> 

So, basically, even if I just copy a config file which used to work and train (without tokens_per_instance) I now get this error. 
I know some fields do not make sense, this is just the result of various tries.
The environment is new and clean, and I tried reinstalling the nvidia drivers. 

Is this an hardware issue?
Thank you in advance

**System :**
 - OS: [Linux]
 - Python version: [3.6.8]
 - AllenNLP version: [I installed from master]
 - PyTorch version: 1.0.1.post2

**Comment by brendan-ai2** [comment]
Glad you figured everything out, @Rainelz!
